# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.h
# Commit: 6770fb6b8d92
# Full Hash: 6770fb6b8d9217b7d75ed7169cb1389c868498ed
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2024-09-06 21:37:32
# Regressor Bug: 1915596
# File Overlap Count: 2
# Description:
#   Bug 1915596 [Wayland] Get touchpad hold event directly from Wayland display r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220551
# ==============================================================================

diff -r 791b8e439bbd -r 6770fb6b8d92 widget/gtk/nsWaylandDisplay.h
--- a/widget/gtk/nsWaylandDisplay.h	Fri Sep 06 08:37:00 2024 +0000
+++ b/widget/gtk/nsWaylandDisplay.h	Fri Sep 06 09:16:28 2024 +0000
@@ -16,6 +16,7 @@
 #include "mozilla/widget/idle-inhibit-unstable-v1-client-protocol.h"
 #include "mozilla/widget/relative-pointer-unstable-v1-client-protocol.h"
 #include "mozilla/widget/pointer-constraints-unstable-v1-client-protocol.h"
+#include "mozilla/widget/pointer-gestures-unstable-v1-client-protocol.h"
 #include "mozilla/widget/linux-dmabuf-unstable-v1-client-protocol.h"
 #include "mozilla/widget/viewporter-client-protocol.h"
 #include "mozilla/widget/xdg-activation-v1-client-protocol.h"
@@ -57,7 +58,20 @@
   }
   bool IsPrimarySelectionEnabled() { return mIsPrimarySelectionEnabled; }
 
+  wl_pointer* GetPointer() { return mPointer; }
+  void SetPointer(wl_pointer* aPointer);
+  void RemovePointer();
+
   void SetShm(wl_shm* aShm);
+
+  void SetKeyboard(wl_keyboard* aKeyboard);
+  wl_keyboard* GetKeyboard() { return mKeyboard; }
+  void ClearKeyboard();
+
+  void SetSeat(wl_seat* aSeat, int aSeatId);
+  wl_seat* GetSeat() { return mSeat; }
+  void RemoveSeat(int aSeatId);
+
   void SetCompositor(wl_compositor* aCompositor);
   void SetSubcompositor(wl_subcompositor* aSubcompositor);
   void SetDataDeviceManager(wl_data_device_manager* aDataDeviceManager);
@@ -66,6 +80,7 @@
   void SetRelativePointerManager(
       zwp_relative_pointer_manager_v1* aRelativePointerManager);
   void SetPointerConstraints(zwp_pointer_constraints_v1* aPointerConstraints);
+  void SetPointerGestures(zwp_pointer_gestures_v1* aPointerGestures);
   void SetDmabuf(zwp_linux_dmabuf_v1* aDmabuf);
   void SetXdgActivation(xdg_activation_v1* aXdgActivation);
   void SetXdgDbusAnnotationManager(
@@ -84,9 +99,15 @@
   wl_compositor* mCompositor = nullptr;
   wl_subcompositor* mSubcompositor = nullptr;
   wl_shm* mShm = nullptr;
+  wl_seat* mSeat = nullptr;
+  int mSeatId = -1;
+  wl_keyboard* mKeyboard = nullptr;
+  wl_pointer* mPointer = nullptr;
   zwp_idle_inhibit_manager_v1* mIdleInhibitManager = nullptr;
   zwp_relative_pointer_manager_v1* mRelativePointerManager = nullptr;
   zwp_pointer_constraints_v1* mPointerConstraints = nullptr;
+  zwp_pointer_gestures_v1* mPointerGestures = nullptr;
+  zwp_pointer_gesture_hold_v1* mPointerGestureHold = nullptr;
   wp_viewporter* mViewporter = nullptr;
   zwp_linux_dmabuf_v1* mDmabuf = nullptr;
   xdg_activation_v1* mXdgActivation = nullptr;